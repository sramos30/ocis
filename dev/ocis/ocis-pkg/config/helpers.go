package config

import (
	"io/fs"
	"os"
	"path/filepath"
	"strings"

	gofig "github.com/gookit/config/v2"
	gooyaml "github.com/gookit/config/v2/yaml"
)

var (
	defaultLocations = []string{
		filepath.Join(os.Getenv("HOME"), "/.ocis/config/"),
		"/etc/ocis/",
		".config/",
	}

	// supportedExtensions is determined by gookit/config.
	supportedExtensions = []string{
		"yaml",
		"yml",
	}
)

// DefaultConfigSources returns a slice with matched expected config files. It sugars coat several aspects of config file
// management by assuming there are 3 default locations a config file could be.
// It uses globbing to match a config file by name, and retrieve any supported extension supported by our drivers.
// It sanitizes the output depending on the list of drivers provided.
func DefaultConfigSources(filename string, drivers []string) []string {
	var sources []string

	locations := []string{}
	if v := os.Getenv("OCIS_CONFIG_DIR"); v != "" {
	    locations = append(locations, v)
		// only use the configured config dir
		locations = append(locations, os.Getenv("OCIS_CONFIG_DIR"))
	} else {
		// merge config from all default locations
		locations = append(locations, defaultLocations...)
	}

	for i := range locations {
		dirFS := os.DirFS(locations[i])
		pattern := filename + ".*"
		matched, _ := fs.Glob(dirFS, pattern)
		if len(matched) > 0 {
			// prepend path to results
			for j := 0; j < len(matched); j++ {
				matched[j] = filepath.Join(locations[i], matched[j])
			}
		}
		sources = append(sources, matched...)
	}

	return sanitizeExtensions(sources, drivers, func(a, b string) bool {
		return strings.HasSuffix(filepath.Base(a), b)
	})
}

// sanitizeExtensions removes elements from "set" which extensions are not in "ext".
func sanitizeExtensions(set []string, ext []string, f func(a, b string) bool) []string {
	var r []string
	for i := 0; i < len(set); i++ {
		for j := 0; j < len(ext); j++ {
			if f(filepath.Base(set[i]), ext[j]) {
				r = append(r, set[i])
			}
		}
	}
	return r
}

// BindSourcesToStructs assigns any config value from a config file / env variable to struct `dst`. Its only purpose
// is to solely modify `dst`, not dealing with the config structs; and do so in a thread safe manner.
func BindSourcesToStructs(extension string, dst interface{}) (*gofig.Config, error) {
	sources := DefaultConfigSources(extension, supportedExtensions)
	cnf := gofig.NewWithOptions(extension)
	cnf.WithOptions(func(options *gofig.Options) {
		options.DecoderConfig.TagName = "yaml"
	})
	cnf.AddDriver(gooyaml.Driver)
	_ = cnf.LoadFiles(sources...)

	err := cnf.BindStruct("", &dst)
	if err != nil {
		return nil, err
	}

	return cnf, nil
}
