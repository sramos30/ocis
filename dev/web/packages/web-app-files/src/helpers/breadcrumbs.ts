import { bus } from 'web-pkg/src/instance'

export interface BreadcrumbItem {
  text: string
  to?: string
  allowContextActions?: boolean
  onClick?: () => void
}

export const breadcrumbsFromPath = (
  currentPath: string,
  resourcePath: string
): BreadcrumbItem[] => {
  const pathSplit = (p = '') => p.split('/').filter(Boolean)
  const current = pathSplit(currentPath)
  const resource = pathSplit(resourcePath)

  return resource.map(
    (text, i) =>
      ({
        allowContextActions: true,
        text,
        to: '/' + [...current].splice(0, current.length - resource.length + i + 1).join('/')
      } as BreadcrumbItem)
  )
}

export const concatBreadcrumbs = (...items: BreadcrumbItem[]): BreadcrumbItem[] => {
  const last = items.pop()

  return [
    ...items,
    {
      allowContextActions: last.allowContextActions,
      text: last.text,
      onClick: () => bus.publish('app.files.list.load')
    }
  ]
}
