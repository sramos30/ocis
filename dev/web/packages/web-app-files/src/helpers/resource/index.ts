export * from './asset'
export * from './filter'
export * from './privatePreviewBlob'
export * from './publicPreviewUrl'
export * from './resource'
export * from './sameResource'
