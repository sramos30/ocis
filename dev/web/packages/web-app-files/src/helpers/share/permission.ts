// dummy to trick gettext string extraction into recognizing strings
const $gettext = (str) => {
  return str
}

export class SharePermission {
  private readonly _key: string
  private readonly _bit: number
  private readonly _label: string

  constructor(key: string, bit: number, label: string) {
    this._key = key
    this._bit = bit
    this._label = label
  }

  get key(): string {
    return this._key
  }

  get bit(): number {
    return this._bit
  }

  get label(): string {
    return $gettext(this._label)
  }

  public enabled(bitmask: number): boolean {
    return (bitmask & this.bit) > 0
  }
}

export abstract class SharePermissions {
  static readonly read = new SharePermission('read', 1, $gettext('Read'))
  static readonly update = new SharePermission('update', 2, $gettext('Update'))
  static readonly create = new SharePermission('create', 4, $gettext('Create'))
  static readonly delete = new SharePermission('delete', 8, $gettext('Delete'))
  static readonly share = new SharePermission('share', 16, $gettext('Share'))

  static permissionsToBitmask(permissions: SharePermission[]): number {
    return (permissions || []).reduce((b: number, p: SharePermission) => b | p.bit, 0)
  }

  static bitmaskToPermissions(bitmask: number): SharePermission[] {
    return [this.read, this.update, this.create, this.delete, this.share].reduce(
      (a: SharePermission[], p: SharePermission) => {
        if (p.enabled(bitmask)) {
          a.push(p)
        }
        return a
      },
      []
    )
  }
}
