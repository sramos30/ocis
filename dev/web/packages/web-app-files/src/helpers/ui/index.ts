export * as fileList from './filesList'
export * as resourceTable from './resourceTable'
