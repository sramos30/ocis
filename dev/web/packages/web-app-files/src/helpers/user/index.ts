export * from './avatarUrl'
export * from './types'
