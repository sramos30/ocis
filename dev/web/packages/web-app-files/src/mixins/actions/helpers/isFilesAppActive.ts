import { activeApp } from 'web-pkg/src/composables'

const isFilesAppActive = (activeApp: string): boolean => {
  // FIXME: we should use this constant but it somehow breaks the unit tests
  // return activeApp === FilesApp.appInfo.id
  return activeApp === 'files'
}

export default {
  computed: {
    $_isFilesAppActive(): boolean {
      return isFilesAppActive(activeApp(this.$route))
    }
  }
}
