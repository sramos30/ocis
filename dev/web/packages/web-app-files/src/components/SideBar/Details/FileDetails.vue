<template>
  <div id="oc-file-details-sidebar">
    <div v-if="hasContent">
      <div
        v-if="file.thumbnail"
        key="file-thumbnail"
        :style="{
          'background-image': $asyncComputed.preview.updating ? 'none' : `url(${preview})`
        }"
        class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb"
        data-testid="preview"
      >
        <oc-spinner v-if="$asyncComputed.preview.updating" />
      </div>
      <div
        v-if="shareIndicators.length"
        key="file-shares"
        data-testid="sharingInfo"
        class="oc-flex oc-flex-middle oc-my-m"
      >
        <oc-status-indicators :resource="file" :indicators="shareIndicators" />
        <p class="oc-my-rm oc-mx-s" v-text="detailSharingInformation" />
      </div>
      <table class="details-table" :aria-label="detailsTableLabel">
        <tr v-if="hasTimestamp" data-testid="timestamp">
          <th scope="col" class="oc-pr-s" v-text="timestampTitle" />
          <td>
            <oc-button
              v-if="showVersions"
              v-oc-tooltip="seeVersionsLabel"
              appearance="raw"
              :aria-label="seeVersionsLabel"
              @click="expandVersionsPanel"
              v-text="capitalizedTimestamp"
            />
            <span v-else v-text="capitalizedTimestamp" />
          </td>
        </tr>
        <tr v-if="showSharedVia" data-testid="shared-via">
          <th scope="col" class="oc-pr-s" v-text="sharedViaLabel" />
          <td>
            <router-link :to="sharedParentRoute">
              <span v-oc-tooltip="sharedViaTooltip" v-text="sharedParentDir" />
            </router-link>
          </td>
        </tr>
        <tr v-if="showSharedBy" data-testid="shared-by">
          <th scope="col" class="oc-pr-s" v-text="sharedByLabel" />
          <td>
            <span v-text="sharedByDisplayName" />
          </td>
        </tr>
        <tr v-if="ownerDisplayName" data-testid="ownerDisplayName">
          <th scope="col" class="oc-pr-s" v-text="ownerTitle" />
          <td>
            <p class="oc-m-rm">
              {{ ownerDisplayName }}
              <span v-if="ownedByCurrentUser" v-translate>(me)</span>
              <span v-if="!ownedByCurrentUser && ownerAdditionalInfo"
                >({{ ownerAdditionalInfo }})</span
              >
            </p>
          </td>
        </tr>
        <tr v-if="showSize" data-testid="sizeInfo">
          <th scope="col" class="oc-pr-s" v-text="sizeTitle" />
          <td v-text="getResourceSize(file.size)" />
        </tr>
        <tr v-if="showVersions" data-testid="versionsInfo">
          <th scope="col" class="oc-pr-s" v-text="versionsTitle" />
          <td>
            <oc-button
              v-oc-tooltip="seeVersionsLabel"
              appearance="raw"
              :aria-label="seeVersionsLabel"
              @click="expandVersionsPanel"
              v-text="versions.length"
            />
          </td>
        </tr>
      </table>
    </div>
    <p v-else data-testid="noContentText" v-text="noContentText" />
  </div>
</template>
<script lang="ts">
import { computed, defineComponent, ref } from '@vue/composition-api'
import Mixins from '../../../mixins'
import MixinResources from '../../../mixins/resources'
import { mapActions, mapGetters } from 'vuex'
import { ImageDimension } from '../../../constants'
import { loadPreview } from '../../../helpers/resource'
import upperFirst from 'lodash-es/upperFirst'
import path from 'path'
import { createLocationSpaces, isAuthenticatedRoute, isLocationSpacesActive } from '../../../router'
import { ShareTypes } from '../../../helpers/share'
import { useRoute, useRouter } from 'web-pkg/src/composables'
import { getIndicators } from '../../../helpers/statusIndicators'

export default defineComponent({
  name: 'FileDetails',
  mixins: [Mixins, MixinResources],

  inject: ['displayedItem'],
  setup() {
    const sharedParentDir = ref('')
    const router = useRouter()
    const route = useRoute()

    const sharedParentRoute = computed(() => {
      if (isLocationSpacesActive(router, 'files-spaces-project')) {
        return createLocationSpaces('files-spaces-project', {
          params: { storageId: route.value.params.storageId, item: sharedParentDir.value }
        })
      }

      return createLocationSpaces('files-spaces-personal-home', {
        params: { item: sharedParentDir.value }
      })
    })

    return {
      sharedParentDir,
      sharedParentRoute
    }
  },

  data: () => ({
    loading: false,
    sharedByName: '',
    sharedByDisplayName: '',
    sharedTime: 0,
    sharedItem: null,
    shareIndicators: []
  }),
  computed: {
    ...mapGetters('Files', ['versions', 'sharesTree', 'sharesTreeLoading']),
    ...mapGetters(['user', 'getToken', 'configuration']),

    file() {
      return this.displayedItem.value
    },

    hasContent() {
      return (
        this.hasTimestamp ||
        this.ownerDisplayName ||
        this.showSize ||
        this.showShares ||
        this.showVersions
      )
    },
    noContentText() {
      return this.$gettext('No information to display')
    },
    detailsTableLabel() {
      return this.$gettext('Overview of the information about the selected file')
    },
    shareDateLabel() {
      return this.$gettext('Shared:')
    },
    sharedViaLabel() {
      return this.$gettext('Shared via:')
    },
    sharedViaTooltip() {
      return this.$gettextInterpolate(
        this.$gettext("Navigate to '%{folder}'"),
        { folder: this.sharedParentDir || '' },
        true
      )
    },
    showSharedBy() {
      return (
        this.showShares &&
        !this.ownedByCurrentUser &&
        !this.sharesTreeLoading &&
        this.sharedByDisplayName &&
        this.sharedByName !== this.file.ownerId
      )
    },
    showSharedVia() {
      return (
        this.showShares &&
        !this.sharesTreeLoading &&
        this.file.path !== this.sharedParentDir &&
        this.sharedParentDir !== null
      )
    },
    showShares() {
      return this.hasAnyShares && isAuthenticatedRoute(this.$route)
    },
    detailSharingInformation() {
      const isFolder = this.file.type === 'folder'
      if (isFolder) {
        return this.$gettext('This folder has been shared.')
      }
      return this.$gettext('This file has been shared.')
    },
    sharedByLabel() {
      return this.$gettext('Shared by:')
    },
    hasTimestamp() {
      return this.file.mdate?.length > 0
    },
    timestampTitle() {
      return this.$gettext('Last modified:')
    },
    ownerTitle() {
      return this.$gettext('Owner:')
    },
    ownerDisplayName() {
      return (
        this.file.ownerDisplayName ||
        this.file.shareOwnerDisplayname ||
        this.file.owner?.[0].displayName
      )
    },
    ownerAdditionalInfo() {
      return this.file.owner?.[0].additionalInfo
    },
    showSize() {
      return this.getResourceSize(this.file.size) !== '?'
    },
    sizeTitle() {
      return this.$gettext('Size:')
    },
    showVersions() {
      if (this.file.type === 'folder') {
        return
      }
      return this.versions.length > 0 && isAuthenticatedRoute(this.$route)
    },
    versionsTitle() {
      return this.$gettext('Versions:')
    },
    seeVersionsLabel() {
      return this.$gettext('See all versions')
    },
    capitalizedTimestamp() {
      const displayDate = this.formDateFromHTTP(this.file.mdate)
      return upperFirst(displayDate)
    },
    hasAnyShares() {
      return (
        this.file.shareTypes?.length > 0 ||
        this.file.indicators?.length > 0 ||
        this.sharedItem !== null
      )
    },
    ownedByCurrentUser() {
      return (
        this.file.ownerId === this.user.id ||
        this.file.owner?.[0].username === this.user.id ||
        this.file.shareOwner === this.user.id
      )
    }
  },
  watch: {
    file() {
      this.loadData().then(this.refreshShareDetailsTree)
    },
    sharesTree() {
      // missing early return
      this.sharedItem = null
      this.shareIndicators = getIndicators(this.file, this.sharesTree)
      const sharePathParentOrCurrent = this.getParentSharePath(this.file.path, this.sharesTree)
      if (sharePathParentOrCurrent === null) {
        return
      }
      const userShares = this.sharesTree[sharePathParentOrCurrent]?.filter((s) =>
        ShareTypes.containsAnyValue(ShareTypes.individuals, [s.shareType])
      )
      if (userShares.length === 0) {
        return
      }

      this.sharedItem = userShares[0]
      this.sharedByName = this.sharedItem.owner?.name
      this.sharedByDisplayName = this.sharedItem.owner?.displayName
      if (this.sharedItem.owner?.additionalInfo) {
        this.sharedByDisplayName += ' (' + this.sharedItem.owner.additionalInfo + ')'
      }
      this.sharedTime = this.sharedItem.stime
      this.sharedParentDir = sharePathParentOrCurrent
    }
  },
  mounted() {
    this.loadData()
    this.refreshShareDetailsTree()
  },
  asyncComputed: {
    preview: {
      async get() {
        // TODO: this timeout resolves flickering of the preview because it's rendered multiple times. Needs a better solution.
        await new Promise((resolve) => setTimeout(resolve, 500))
        return loadPreview({
          resource: this.file,
          isPublic: !isAuthenticatedRoute(this.$route),
          dimensions: ImageDimension.Preview,
          server: this.configuration.server,
          userId: this.user.id,
          token: this.getToken
        })
      },
      lazy: true,
      watch: ['file.id']
    }
  },
  methods: {
    ...mapActions('Files', ['loadPreview', 'loadVersions', 'loadSharesTree']),
    ...mapActions('Files/sidebar', { setSidebarPanel: 'setActivePanel' }),
    async refreshShareDetailsTree() {
      await this.loadSharesTree({
        client: this.$client,
        path: this.file.path,
        $gettext: this.$gettext,
        storageId: this.$route.params.storageId
      })
      this.shareIndicators = getIndicators(this.file, this.sharesTree)
    },
    getParentSharePath(childPath, shares) {
      let currentPath = childPath
      if (!currentPath) {
        return null
      }
      while (currentPath !== '/') {
        const share = shares[currentPath]
        if (share !== undefined && share[0] !== undefined) {
          return currentPath
        }
        currentPath = path.dirname(currentPath)
      }
      return null
    },
    expandPeoplesPanel() {
      this.setSidebarPanel('sharing-item')
    },
    expandLinksPanel() {
      this.setSidebarPanel('links-item')
    },
    expandVersionsPanel() {
      this.setSidebarPanel('versions-item')
    },
    async loadData() {
      this.loading = true
      const calls = []
      if (this.file.type === 'file' && isAuthenticatedRoute(this.$route)) {
        calls.push(this.loadVersions({ client: this.$client, fileId: this.file.id }))
      }
      await Promise.all(calls.map((p) => p.catch((e) => e)))
      this.loading = false
    }
  }
})
</script>
<style lang="scss" scoped>
.details-table {
  text-align: left;

  tr {
    height: 1.5rem;
  }

  th {
    font-weight: 600;
  }
}

.details-preview {
  background-color: var(--oc-color-background-muted);
  border: 10px solid var(--oc-color-background-muted);
  height: 230px;

  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
}
</style>
