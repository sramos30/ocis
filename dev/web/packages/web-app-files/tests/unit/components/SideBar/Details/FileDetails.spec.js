import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import FileDetails from '../../../../../src/components/SideBar/Details/FileDetails.vue'
import stubs from '../../../../../../../tests/unit/stubs'
import GetTextPlugin from 'vue-gettext'
import AsyncComputed from 'vue-async-computed'
import { ShareTypes } from '../../../../../src/helpers/share'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(AsyncComputed)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})
const OcTooltip = jest.fn()

const simpleOwnFolder = {
  id: '1',
  type: 'folder',
  ownerId: 'marie',
  ownerDisplayName: 'Marie',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740'
}

const sharedFolder = {
  id: '2',
  type: 'folder',
  ownerId: 'einstein',
  ownerDisplayName: 'Einstein',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740',
  shareTypes: [ShareTypes.user.value]
}

const simpleOwnFile = {
  id: '3',
  type: 'file',
  ownerId: 'marie',
  ownerDisplayName: 'Marie',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740'
}

const sharedFile = {
  id: '4',
  path: '/Shares/123.png',
  type: 'file',
  ownerId: 'einstein',
  ownerDisplayName: 'Einstein',
  preview: 'example.com/image',
  thumbnail: 'example.com/image',
  mdate: 'Tue, 20 Oct 2015 06:15:00 GMT',
  size: '740',
  shareTypes: [ShareTypes.user.value]
}

const formDateFromJSDate = jest.fn().mockImplementation(() => 'ABSOLUTE_TIME')
const formDateFromHTTP = jest.fn().mockImplementation(() => 'ABSOLUTE_TIME')
const refreshShareDetailsTree = jest.fn()

beforeEach(() => {
  formDateFromJSDate.mockClear()
  formDateFromHTTP.mockClear()
  refreshShareDetailsTree.mockReset()
})

describe('Details SideBar Panel', () => {
  describe('displays a resource of type folder', () => {
    describe('on a private page', () => {
      it('with timestamp, size info and (me) as owner', () => {
        const wrapper = createWrapper(simpleOwnFolder)
        expect(wrapper).toMatchSnapshot()
      })
      it('with timestamp, size info, share info and share date', () => {
        const wrapper = createWrapper(sharedFolder)
        expect(wrapper).toMatchSnapshot()
      })
    })
    describe('on a public page', () => {
      it('with owner, timestamp, size info and no share info', () => {
        const wrapper = createWrapper(sharedFolder, [], null, true)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
  describe('displays a resource of type file', () => {
    describe('on a private page', () => {
      it('with timestamp, size info and (me) as owner', () => {
        const wrapper = createWrapper(simpleOwnFile)
        expect(wrapper).toMatchSnapshot()
      })
      it('with timestamp, size info, share info, share date and preview', () => {
        const wrapper = createWrapper(sharedFile)
        expect(wrapper).toMatchSnapshot()
      })

      it('updates when the shareTree updates', async () => {
        const wrapper = createWrapper(sharedFile)
        expect(refreshShareDetailsTree).toBeCalledTimes(1)

        // make sure this renders once when initial sharesTree become available
        wrapper.vm.$store.state.Files.sharesTree = {
          '/Shares': [{}]
        }

        await wrapper.vm.$nextTick
        expect(wrapper).toMatchSnapshot()

        // ... and renders again when the relevant shares become available
        wrapper.vm.$store.state.Files.sharesTree = {
          '/Shares': [{}],
          '/Shares/123.png': [
            {
              shareType: 0,
              owner: {
                name: 'marie',
                displayName: 'Marie Curie'
              },
              stime: 12345
            }
          ]
        }
        await wrapper.vm.$nextTick
        expect(wrapper).toMatchSnapshot()
      })
    })
    describe('on a public page', () => {
      it('with owner, timestap, size info, no share info and preview', () => {
        const wrapper = createWrapper(sharedFile, [], null, true)
        expect(wrapper).toMatchSnapshot()
      })
      it('with owner, timestamp, size info, no share info and preview', () => {
        const wrapper = createWrapper(sharedFile, [], null, true)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})

function createWrapper(testResource, testVersions = [], testPreview, publicRoute = false) {
  return shallowMount(FileDetails, {
    store: new Vuex.Store({
      getters: {
        user: function () {
          return { id: 'marie' }
        }
      },
      modules: {
        Files: {
          namespaced: true,
          state: {
            sharesTree: {}
          },
          getters: {
            highlightedFile: function () {
              return testResource
            },
            versions: function () {
              return 2
            },
            sharesTreeLoading: function () {
              return false
            },
            sharesTree: function (state) {
              return state.sharesTree
            }
          },
          actions: {
            loadVersions: function () {
              return testVersions
            },
            loadPreview: function () {
              return testPreview
            },
            loadSharesTree: jest.fn()
          }
        }
      }
    }),
    localVue,
    stubs: stubs,
    directives: {
      OcTooltip
    },
    mixins: [
      {
        methods: {
          formDateFromJSDate,
          formDateFromHTTP,
          refreshShareDetailsTree
        }
      }
    ],
    mocks: {
      $route: {
        meta: {
          auth: !publicRoute
        }
      }
    },
    provide: {
      displayedItem: {
        value: testResource
      }
    }
  })
}
