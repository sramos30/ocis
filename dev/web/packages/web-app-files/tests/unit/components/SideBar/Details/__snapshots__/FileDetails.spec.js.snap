// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details SideBar Panel displays a resource of type file on a private page updates when the shareTree updates 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]"></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <tr data-testid="shared-via">
        <th scope="col" class="oc-pr-s">Shared via:</th>
        <td>
          <router-link-stub><span></span></router-link-stub>
        </td>
      </tr>
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page updates when the shareTree updates 2`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]"></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <tr data-testid="shared-via">
        <th scope="col" class="oc-pr-s">Shared via:</th>
        <td>
          <router-link-stub><span>/Shares/123.png</span></router-link-stub>
        </td>
      </tr>
      <tr data-testid="shared-by">
        <th scope="col" class="oc-pr-s">Shared by:</th>
        <td><span>Marie Curie</span></td>
      </tr>
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page with timestamp, size info and (me) as owner 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <!---->
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Marie
            <span data-msgid="(me)" data-current-language="en_US">(me)</span>
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page with timestamp, size info, share info, share date and preview 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub></oc-spinner-stub>
    </div>
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <tr data-testid="shared-via">
        <th scope="col" class="oc-pr-s">Shared via:</th>
        <td>
          <router-link-stub><span></span></router-link-stub>
        </td>
      </tr>
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a public page with owner, timestamp, size info, no share info and preview 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub></oc-spinner-stub>
    </div>
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a public page with owner, timestap, size info, no share info and preview 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub></oc-spinner-stub>
    </div>
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a private page with timestamp, size info and (me) as owner 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <!---->
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Marie
            <span data-msgid="(me)" data-current-language="en_US">(me)</span>
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a private page with timestamp, size info, share info and share date 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <!---->
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a public page with owner, timestamp, size info and no share info 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <!---->
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified:</th>
        <td><span>ABSOLUTE_TIME</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner:</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size:</th>
        <td>740 B</td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;
